gk/main.c

```diff
@@ -1251,9 +1251,9 @@ xmit_icmpv6(struct gatekeeper_if *iface, struct ipacket *packet,
 	 * RFC 4861, sections 7.1.1 and 7.1.2.
 	 */
 	icmp_ipv6->hop_limits = 255;
-	rte_memcpy(icmp_ipv6->src_addr, packet->flow.f.v6.dst,
+	rte_memcpy(icmp_ipv6->src_addr, packet->flow.f.v6.dst.s6_addr,
 		sizeof(icmp_ipv6->src_addr));
-	rte_memcpy(icmp_ipv6->dst_addr, packet->flow.f.v6.src,
+	rte_memcpy(icmp_ipv6->dst_addr, packet->flow.f.v6.src.s6_addr,
```

Comment:
Use type safe assignments here as well.

Reply:
The field ``src_addr`` in ipv6 header is defined as ``uint8_t src_addr [16]`` instead of the type ``struct in6_addr``, so we may not apply the type assignment here.